app/controllers/posts_controller.rb

```diff
@@ -2,21 +2,23 @@ class PostsController < ApplicationController
   before_filter :authorize
   before_action :correct_user, only: [:edit, :update, :destroy]
 
-
   def index
-    @posts = Post.all
+    @team = Team.find(params[:team_id])
+    @posts = @team.posts
   end
 
   def new
-    @post = Post.new
+    @team = Team.find(params[:team_id])
+    @post = @team.posts.new
```

Comment:
You can build your association

http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html